processors/base/writers/location.py

```diff
@@ -70,3 +76,37 @@ def write_location(conn, location, source_id, trial_id=None):
             'created' if create else 'updated', name)
 
     return object['id']
+
+#----helper function to get the canonical name of the locations
+def normalize_location(location):
+    location_name = location.lower().strip().replace(".", "")
+    try:
+        canonical_name = countries.get(location_name).name
+        logger.debug('Location - %s: %s normalized', location_name, canonical_name)
+    except KeyError:
+        canonical_name = location
+        # This part is for looking for matching cases (suggestion), uses the 
+        # fuzzy logic to get the country name who get 80% of match rate.
+        with open(os.path.join(os.path.dirname(__file__),'countries.csv'), 'r') as f:
+            countries_data = csv.reader(f, delimiter= str(u','))
+            logger.debug('Location - %s: %s not normalized', location_name, canonical_name)
+            best_similarity = 80
+            for country in countries_data:
+                unicode_row = [x.decode('utf8') for x in country]
+                country_name = unicode_row[0].lower().strip().replace(".", "")
+                similarity = fuzz.ratio(location_name, country_name)
+                if similarity >= best_similarity:
+                    best_similarity = similarity
+                    canonical_name = countries.get(unicode_row[3]).name
+                    logger.debug('Location - %s: %s not normalized, using the most similar name',
+                     location_name, canonical_name)
+                else:
+                    capital = unicode_row[21].lower().strip().replace(".", "")
+                    similarity = fuzz.ratio(location_name, capital)
+                    if similarity >= best_similarity:
+                        best_similarity = similarity
+                        canonical_name = countries.get(unicode_row[3]).name
+                        logger.debug('Location - %s: %s not normalized, using the most similar capital name',
+                        location_name, canonical_name)
```

Comment:
Instead of using this loop, you can calculate all similarities at once doing something like:

```python
_max_index_and_value = lambda arr: max(enumerate(arr), key=lambda x: x[1])
similarity_with_countries = [fuzz.ratio(location_name, country['name']) for country in countries]
max_index, max_value = _max_index_and_value(similarity_with_countries)
if max_value >= THRESHOLD:
    result = countries.get(countries[max_index]['ISO3166-1-Alpha-2']).name
else:
    # Try capitals
```

This could even be extracted to a separate function, as it's done for capitals and countries. It might help improve the readability, or maybe not. Try to see which you prefer.